src/plugins/resolver/resolver.c

```diff
@@ -370,6 +375,37 @@ int ELEKTRA_PLUGIN_FUNCTION(resolver, open)
 	resolverInit (&p->user, path);
 	resolverInit (&p->system, path);
 
+#if defined(ELEKTRA_LOCK_MUTEX) && defined(ELEKTRA_RESOLVER_RECURSIVE_MUTEX_INITIALIZATION)
+	// PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP is available in glibc only
+	// so we use another mutex for the initialization of the recursive mutex,
+	// since this section must be thread safe.
+	pthread_mutex_lock(&elektra_resolver_init_mutex);
+	if (!elektra_resolver_mutex_initialized)
+	{
+		pthread_mutexattr_t mutex_attr;
+
+		if (pthread_mutexattr_init (&mutex_attr))
+		{
+			ELEKTRA_SET_ERROR(35, errorKey, "Could not initialize recursive mutex");
```

Comment:
the error message should contain where the error happened (attr_init, settype, init), and ideally also the error code.